File: packages/python-sdk/strait/operations/health.py
"""Health check operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class HealthService(BaseService):
    def list(self) -> dict[str, Any]:
        return self._request("GET", "/health")

    def get_ready(self) -> dict[str, Any]:
        return self._request("GET", "/health/ready")

    def list_metrics(self) -> dict[str, Any]:
        return self._request("GET", "/metrics")


class AsyncHealthService(AsyncBaseService):
    async def list(self) -> dict[str, Any]:
        return await self._request("GET", "/health")

    async def get_ready(self) -> dict[str, Any]:
        return await self._request("GET", "/health/ready")

    async def list_metrics(self) -> dict[str, Any]:
        return await self._request("GET", "/metrics")

File: packages/ruby-sdk/lib/strait/operations/health.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for health-check endpoints.
    class HealthService < BaseService
      # List overall health status.
      # GET /v1/health
      def list
        _request(:get, "/v1/health")
      end

      # Check readiness probe.
      # GET /v1/health/ready
      def get_ready
        _request(:get, "/v1/health/ready")
      end

      # List runtime metrics.
      # GET /v1/health/metrics
      def list_metrics
        _request(:get, "/v1/health/metrics")
      end
    end
  end
end
